aptos-move/block-executor/src/executor.rs

```diff
@@ -623,14 +606,106 @@ where
             // TODO: re-use the code cache.
             drop(versioned_cache);
         });
+        ret
+    }
 
-        match maybe_err {
-            Some(err) => Err(err),
-            None => {
-                final_results.resize_with(num_txns, E::Output::skip_output);
-                Ok(final_results)
-            },
+    fn set_error_if_needed(
+        maybe_err: &Arc<Mutex<Option<Error<E::Error>>>>,
+        min_error_index: &Arc<Mutex<Option<usize>>>,
+        err: Error<E::Error>,
+        idx: usize,
+    ) {
+        let mut min_error_index = min_error_index.lock().unwrap();
+        if let Some(existing_index) = *min_error_index {
+            if idx < existing_index {
+                *maybe_err.lock().unwrap() = Some(err);
```

Comment:
The `min_error_index` mutex already guards the `maybe_err` - the second lock is unnecessary. Let's get around Rust's limits in a different way (use a tuple guarded by a mutex?).